code/Controllers/CMSMain.php

```diff
@@ -1666,62 +1534,21 @@ public function childfilter(HTTPRequest $request): HTTPResponse
             ->setBody(json_encode($disallowedChildren));
     }
 
-    /**
-     * Safely reconstruct a selected filter from a given set of query parameters
-     *
-     * @param array $params Query parameters to use, or null if none present
-     * @return CMSSiteTreeFilter The filter class
-     * @throws InvalidArgumentException if invalid filter class is passed.
-     * @deprecated 5.4.0 Will be removed without equivalent functionality to replace it.
-     */
-    protected function getQueryFilter($params)
-    {
-        Deprecation::noticeWithNoReplacment('5.4.0');
-        if (empty($params['FilterClass'])) {
-            return null;
-        }
-        $filterClass = $params['FilterClass'];
-        if (!is_subclass_of($filterClass, CMSSiteTreeFilter::class)) {
-            throw new InvalidArgumentException("Invalid filter class passed: {$filterClass}");
-        }
-        return $filterClass::create($params);
-    }
-
-    /**
-     * Returns the records meet a certain criteria as {@see CMSSiteTreeFilter} or the subrecords of a parent record
-     * defaulting to no filter and show all records in first level.
-     * Doubles as search results, if any search parameters are set through {@link SearchForm()}.
-     *
-     * @param array $params Search filter criteria
-     * @param int $parentID Optional parent node to filter on (can't be combined with other search criteria)
-     * @return SS_List
-     * @throws InvalidArgumentException if invalid filter class is passed.
-     * @deprecated 5.4.0 Will be removed without equivalent functionality to replace it.
-     */
-    public function getList($params = [], $parentID = 0)
-    {
-        Deprecation::noticeWithNoReplacment('5.4.0');
-        if ($filter = $this->getQueryFilter($params)) {
-            return $filter->getFilteredPages();
-        } else {
-            $list = DataObject::get($this->getModelClass());
-            $parentID = is_numeric($parentID) ? $parentID : 0;
-            return $list->filter("ParentID", $parentID);
-        }
-    }
-
     /**
      * @return Form
      */
     public function ListViewForm()
     {
-        $params = $this->getRequest()->requestVar('q');
+        $modelClass = $this->getModelClass();
         $parentID = $this->getRequest()->requestVar('ParentID');
-        // Set default filter if other params are set
-        if ($params && empty($params['FilterClass'])) {
-            $params['FilterClass'] = CMSSiteTreeFilter_Search::class;
+        $params = $this->getRequest()->requestVar('q') ?? [];
+        if ($params) {
+            $form = $this->getSearchForm();
+            $params = $form->prepareValuesForSearchContext($params);
+            $list = DataObject::singleton($modelClass)->getDefaultSearchContext()->getQuery($params);
+        } else {
+            $list = DataObject::singleton($modelClass)::get()->filter('ParentID', is_numeric($parentID) ? $parentID : 0);
```

Comment:
We should be instead throwing an exception if $parentID isn't numeric? There's no reason for it to be non-numeric?

Reply:
I'm just using the same logic that was already in place. https://github.com/silverstripe/silverstripe-cms/blob/c989e4f39bc5fad3527875ae3fce1502dfdcec1b/code/Controllers/CMSMain.php#L1812
Changing that isn't in scope - and given the short time til the beta I'd prefer not to be making changes we don't need to.